src/biogeophys/HillslopeHydrologyMod.F90

```diff
@@ -0,0 +1,1233 @@
+
+    do c = bounds%begc, bounds%endc
+       if (col%is_hillslope_column(c)) then
+          ! In preparation for this re-weighting of patch type
+          ! only first patch was given a non-zero weight in surfrd_hillslope
+          npatches_per_column = 0
+          do p = col%patchi(c), col%patchf(c)
+             patch%itype(p) = col_pftndx(c)
+             ! update mxy as is done in initSubgridMod.add_patch
+             patch%mxy(p) = patch%itype(p) + (1 - natpft_lb)
+             npatches_per_column = npatches_per_column + 1
+          enddo
+          if (check_npatches) then
+             if ((npatches_per_column /= 1) .and. masterproc) then
+                call endrun( 'ERROR:: number of patches per hillslope column not equal to 1'//errmsg(sourcefile, __LINE__) )
+             end if
+          endif
+       endif
+    enddo
+
+  end subroutine HillslopePftFromFile
+
+  !-----------------------------------------------------------------------
+  subroutine HillslopeStreamOutflow(bounds, &
+       waterstatebulk_inst, waterfluxbulk_inst,streamflow_method)
+    !
+    ! !DESCRIPTION:
+    ! Calculate discharge from stream channel
+    !
+    ! !USES:
+    use LandunitType    , only : lun                
+    use GridcellType    , only : grc                
+    use ColumnType      , only : col                
+    use WaterFluxBulkType   , only : waterfluxbulk_type
+    use WaterStateBulkType  , only : waterstatebulk_type
+    use spmdMod         , only : masterproc
+    use clm_varcon      , only : spval, ispval, grlnd 
+    use landunit_varcon , only : istsoil
+    use ncdio_pio
+    use clm_time_manager , only : get_step_size_real
+     
+    !
+    ! !ARGUMENTS:
+    type(bounds_type), intent(in) :: bounds
+    integer,  intent(in)  :: streamflow_method
+    type(waterstatebulk_type), intent(inout) :: waterstatebulk_inst
+    type(waterfluxbulk_type),  intent(inout) :: waterfluxbulk_inst
+
+    integer               :: c, l, g, i, j
+    integer               :: nstep
+    real(r8) :: dtime                                     ! land model time step (sec)
+    real(r8)              :: cross_sectional_area         ! cross sectional area of stream water (m2)
+    real(r8)              :: stream_depth                 ! depth of stream water (m)
+    real(r8)              :: hydraulic_radius             ! cross sectional area divided by wetted perimeter (m)
+    real(r8)              :: flow_velocity                ! flow velocity (m/s)
+    real(r8)              :: overbank_area                ! area of water above bankfull (m2)
+    real(r8), parameter   :: manning_roughness = 0.03_r8  ! manning roughness
+    real(r8), parameter   :: manning_exponent  = 0.667_r8 ! manning exponent
+
+    integer, parameter    :: overbank_method = 1          ! method to treat overbank stream storage; 1 = increase dynamic slope, 2 = increase flow area cross section, 3 = remove instantaneously
```

Comment:
Should `overbank_method` be a namelist parameter?

Reply:
Via [Sean](https://github.com/ESCOMP/CTSM/pull/1715#issuecomment-1904411085):
> these methods are still 'experimental', so we decided to leave the method
local.